cmd/kube-controller-manager/app/certificates.go

```diff
@@ -44,7 +44,12 @@ func startCSRSigningController(ctx ControllerContext) (http.Handler, bool, error
 		klog.Warningf("Resource %s is not available now", gvr.String())
 		return nil, false, nil
 	}
-	if ctx.ComponentConfig.CSRSigningController.ClusterSigningCertFile == "" || ctx.ComponentConfig.CSRSigningController.ClusterSigningKeyFile == "" {
+	hasSingleSigningFile := ctx.ComponentConfig.CSRSigningController.ClusterSigningCertFile == "" || ctx.ComponentConfig.CSRSigningController.ClusterSigningKeyFile == ""
+	hasKubeletServingFile := ctx.ComponentConfig.CSRSigningController.KubeletServingSignerConfiguration.CertFile == "" || ctx.ComponentConfig.CSRSigningController.KubeletServingSignerConfiguration.KeyFile == ""
+	hasKubeletClientFile := ctx.ComponentConfig.CSRSigningController.KubeletClientSignerConfiguration.CertFile == "" || ctx.ComponentConfig.CSRSigningController.KubeletClientSignerConfiguration.KeyFile == ""
+	hasKubeAPIServerClientFile := ctx.ComponentConfig.CSRSigningController.KubeAPIServerClientSignerConfiguration.CertFile == "" || ctx.ComponentConfig.CSRSigningController.KubeAPIServerClientSignerConfiguration.KeyFile == ""
+	hasLegacyUnknownFile := ctx.ComponentConfig.CSRSigningController.LegacyUnknownSignerConfiguration.CertFile == "" || ctx.ComponentConfig.CSRSigningController.LegacyUnknownSignerConfiguration.KeyFile == ""
+	if !hasSingleSigningFile && !hasKubeletServingFile && !hasKubeletClientFile && !hasKubeAPIServerClientFile && !hasLegacyUnknownFile {
```

Comment:
Er, what happens if you specify one of the signers but not the default signing file?

Can we add some tests to exercise this wiring?

Reply:
> Er, what happens if you specify one of the signers but not the default signing file?

You'll get an error about an unspecified file.  Comes from the code doing the initial load.  What is your preferred behavior?

Reply:
> Can we add some tests to exercise this wiring?

Do you have an example of flag wiring behavior tests you like?  I don't object, I just haven't seen many.

Reply:
> You'll get an error about an unspecified file. Comes from the code doing the initial load. What is your preferred behavior?

That seems fine to me.

> Do you have an example of flag wiring behavior tests you like?

There are some tests in `kube-controller-manager/app/options/options_test.go` that test the input flags to `CSRSigningControllerOptions` bits.  Maybe combine that with `startCSRSigningController` and some permutations of the flags?